synapse/cores/ram.py

```diff
@@ -46,6 +49,17 @@ def _sizeByLe(self, prop, valu, limit=None):
     def _rowsByLe(self, prop, valu, limit=None):
         return [ r for r in self.rowsbyprop.get(prop,()) if isint(r[2]) and r[2] <= valu ][:limit]
 
+    def _sizeByIn(self, prop, valu, limit=None):
+        lookup = set(valu)
+        return len([ r for r in self.rowsbyprop.get(prop,()) if r[2] in lookup ])
```

Comment:
size by routine should never return a size which is larger than limit. this implementation could.